### PR TITLE
Add clean_text function family.

### PR DESCRIPTION
**Pull request by jonthegeek**
Split into separate functions for the steps of the cleaning process, but then clean_text wraps them all.

### CODE REVIEW

**Comment by jonthegeek** [comment]
Closes #4.

File: R/clean.R [hunk 1]
```diff
@@ -18,26 +18,211 @@
...
+  # Make sure those ones are explicitly set.
+  Encoding(text[in_encoding_status]) <- "UTF-8"
+
+  # Now try to coerce the leftovers to UTF-8.
+  text[!in_encoding_status] <- purrr::map_chr(
+    text[!in_encoding_status],
+    function(this_text) {
+      converted <- enc2utf8(this_text)
+      if (converted != this_text) {
+        converted <- iconv(text, "latin1", "UTF-8")
```
Comment (jonathanbratt) [inline R/clean.R:49]: Shouldn't this be acting on `this_text` here, and not `text`?

File: R/clean.R [hunk 2]
```diff
@@ -18,26 +18,211 @@
...
+  return(
+    stringi::stri_replace_all_regex(
+      stringi::stri_trans_nfd(text),
+      "\\p{Mn}",
+      ""
+    )
+  )
+}
+
+#' Title
```
Comment (jonathanbratt) [inline R/clean.R:174]: Title and description here?

File: R/clean.R (hunk 1 above)
Comment (jonathanbratt) [reply R/clean.R:49]: This line and the next ~9 are the only lines not covered by tests. :-D

**Review by jonathanbratt — CHANGES_REQUESTED** [review]
just a couple things

**Review by jonathanbratt — APPROVED** [review]
looks good!

**Pull request merged**